elasticsearch/includes/classes/class-elasticsearch.php

```diff
@@ -85,15 +85,23 @@ protected function load_commands() {
 	}
 
 	protected function setup_healthchecks() {
+		$healhcheck = new HealthJob();
+
+		// Disable healthcheck job
+		if ( defined( 'DISABLE_VIP_SEARCH_HEALTHCHECKS' ) && true === DISABLE_VIP_SEARCH_HEALTHCHECKS ) {
+			// Hook into init action to ensure cron-control has already been loaded
+			add_action( 'init', [ $healhcheck, 'disable_job' ] );
```

Comment:
Instead of scheduling this on `init` (which will run on every request), what if we move it into the job itself? So when the job runs, we check the constant, and if it's `true`, we disable the job and return early. So worst case, if it's already scheduled then the constant is added, the job runs 1 additional time but only unregisters itself and returns.

I'm just wary of adding additional code / processing to every request when it only ever needs to happen at most once.

Reply:
For completeness we should wrap the "is enabled" logic in a reusable function that gets called here and once again when the job is running, so that we can bail early.

Currently, disabling the job via the filter won't actually do anything if it's already been scheduled.

Reply:
> Instead of scheduling this on `init` (which will run on every request), what if we move it into the job itself? So when the job runs, we check the constant, and if it's true, we disable the job and return early.

So I tried this and it doesn't seem to be working. It's not removing the existing event. It doesn't run the job though